app/utils/thread/fetch_thread_state.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useState} from 'react';
+import {BehaviorSubject} from 'rxjs';
+
+type State = {[id: string]: boolean};
+
+const defaultState: State = {};
+
+const subject: BehaviorSubject<State> = new BehaviorSubject(defaultState);
+
+export const setFetchThreadState = (rootId: string, status: boolean) => {
+    const prevState = subject.value;
+    subject.next({
+        ...prevState,
+        [rootId]: status,
+    });
+};
+
+export const useFetchThreadState = () => {
+    const [state, setState] = useState(defaultState);
+    useEffect(() => {
+        const sub = subject.subscribe(setState);
+
+        return () => sub.unsubscribe();
+    }, []);
+
+    return state;
+};
```

Comment:
this is nice, but perhaps you want to add a hook for a specific rootId ?

something like
```typescript
export const useFetchThreadStateForRoot = (rootId: string) => {
    const [isFetching, setIsFetching] = useState(false);
    useEffect(() => {
        const sub = subject.pipe(
            switchMap((s) => s[rootId] || false),
            distinctUntilChanged(),
        ).subscribe(setIsFetching)

        return () => sub.unsubscribe();
    }, []);

    return state;
};
```

Reply:
Also, this one would make sense to move to the `hooks` folder while the Behavior can be moved to the `store` folder? that way is easy to find and maintain?

@larkox what do you think if we move the fetching posts event to something like this? I think it'll be less racy

Reply:
Sure! @anurag6713 This has been my approach, which is similar to other similar subjects in the managers.
https://github.com/mattermost/mattermost-mobile/pull/6872
If you want, we can sync to see how we can make it more similar.